Config/OrchestratorMessages.ocfg

```diff
@@ -231,10 +231,12 @@
 
 800(I) : "All composer information cleared."
 801(W) : "Unable to reset a portion of the composer. Did you mean 'clear'?"
-802(W) : "Graph instance '%s' has not been placed. Not %s."
+802(W) : "Graph instance '%s' has not been %s. Not %s."
```

Comment:
Interesting, not sure how I missed that one in my search, but is that POST a mistake anyway? 

Apps_t.cpp:L37 would not have interacted nicely with the old version anyway. I am not sure which message it is meant to be.

Reply:
Scrolling all the way back to `a133b17`, the 802 message was (from the pbuilder section):

```
802(D) : "Removing task %s from the Orchestrator"
```

I think this message is now wrong (we deal with applications, not tasks), and is also missing from both `1.0.0-alpha` and your candidate branch.

I've pushed a change to address this (please check).

Reply:
> Attached is a review of your documentation. No showstoppers, but still worth a gander, I reckon.
> [0172_gmb_documentation_feedback.txt](https://github.com/POETSII/Orchestrator/files/6049413/0172_gmb_documentation_feedback.txt)

>  - Can you refer to the software address documentation in Section 3.1 of your
>   Packet Format documentation? Or is this document supposed to supercede that?

I thought I had done that with the line `software address as described in the Addresses documentation`, but I can clarify further with a more specific statement?

> Occasionally you use verbatim blocks for file names and variables, and
>  occasionally not. I'd recommend making them all verbatim for consistency,
>   but your call.

Meant to be all of them, but I clearly missed a few. Will try to catch some more.


> Page 4: <Insert punchy intro here> is pretty classic, but maybe something a
>   little more formal would be nice? The current intro feels unstructured and a
>  bit "braindumpy" - consider explaining why you're going along the path you
> have set for the reader.

Honestly, I was fed up with the doc when I got there so didn't write an intro and figured it was better to get the meat out there. I will write an intro.


>  - Page 6: Jam in a reference to Figures 3 and 4 here - better flow.

They are already referenced at the bottom of page 6?a

>  - Page 9: The buffering mode softswitch diagram doesn't appear to call OnSend
>   anywhere. I wonder if there's a good way to visually show the difference
>   between Figures 3 and 4 - it's currently hard to discern a difference. Maybe
>   some red text for bits in Figure 4 that are different from Figure 3?

It doesn't have the OnSend in it - this is shown in the RTS sub in the second "Figure 3" (I have fixed the caption and refs). The only difference is that the OnSend block and the following instrumentation poke are replaced with a memcopy. I have added a ref to Fig 5 in the captions of 3 and 4.

>  - Page 10: "softswitch_onRTS() is called any time there may have been a change
>   in a device's state" is not true - as you point out, OnIdle can modify state
>   and return zero, which won't make onRTS get called.

It could be argued that if state is being changed in OnIdle, it should be returning 1. The e.g. kind of clarifies this.

>  - Page 13 (S2.2.3) there are no handlers in Table II. Do you mean macros?

👍 

> - Page 16 "macroin" (maybe run a spell checker).

I did but probably missed a few as it obviously doesn't like things like "struct", "pinAddr", etc. and I am not adding them all to Word's dictionary!

>  - Page 17: Also holds for Tinsel 0.8.3? Also, is there a delay between when
>  the instrumentation statistics are measured and when the packet is sent?

I believe so, but the clock rate seems to be something that may be changed without us being told...

Instrumentation stats are pulled at the time the packet is sent.

>  - Page 18: Is pktBuf in italics for any particular reason?

I think I was trying to make it stand out slightly as something that only exists in buffering mode.




>  - Page 26: Table XIX has some `a`s that should be `x`s.

Good catch!

> Page 29: Empty parentheses after Composer. 

Not sure what they were doing there.

> Table XXI confuses me - is it showing what headers are included for each translation unit? If so, why are some headers "unassigned"?

It is just meant to be a compact list of source and headers that are needed.




>  - Page 37: I'll be honest, I didn't double check this. I'm happy to take your
   word for it. Also, are the boxes supposed to be connected?

No specific connections, as that won't fit on page.



> More generally, I feel like the pacing of this document could be improved by spending more words on
> synopses to introduce concepts, and by shifting some of the reference-manual-tables into appendices.

Yeah, more synopses could be good. I will have a think!

More to come on the linker stuff in a bit.

Reply:
👍